cairo/crates/pragma_dispatcher/src/dispatcher/contract.cairo

```diff
@@ -0,0 +1,295 @@
+    }
+
+    // ================== PUBLIC ABI ==================
+
+    #[abi(embed_v0)]
+    impl PragmaDispatcher of IPragmaDispatcher<ContractState> {
+        /// Returns the registered Pragma Feed Registry address.
+        fn get_pragma_feed_registry_address(self: @ContractState) -> ContractAddress {
+            self.pragma_feed_registry.read().contract_address
+        }
+
+        /// Returns the registered Hyperlane Mailbox address.
+        fn get_hyperlane_mailbox_address(self: @ContractState) -> ContractAddress {
+            self.hyperlane_mailbox.read().contract_address
+        }
+
+        fn get_feed(self: @ContractState, feed_id: FeedId) -> FeedWithId {
+            self.call_get_feed(feed_id)
+        }
+
+        /// Returns the list of supported feeds.
+        fn supported_feeds(self: @ContractState) -> Array<FeedId> {
+            self.call_get_all_feeds()
+        }
+
+        /// Returns the router address registered for an Asset Class.
+        fn get_asset_class_router(
+            self: @ContractState, asset_class_id: AssetClassId
+        ) -> ContractAddress {
+            // [Check] Valid asset class
+            let asset_class: Option<AssetClass> = asset_class_id.try_into();
+            assert(asset_class.is_some(), errors::UNKNOWN_ASSET_CLASS);
+
+            // [Check] A router is registered for the asset class
+            let router = self.asset_class_routers.entry(asset_class.unwrap()).read();
+            assert(!router.is_zero(), errors::NO_ROUTER_REGISTERED);
+
+            // [Interaction] Return the router address
+            router.contract_address
+        }
+
+        /// Register a new router for an Asset Class.
+        fn register_asset_class_router(
+            ref self: ContractState, asset_class_id: AssetClassId, router_address: ContractAddress,
+        ) {
+            // [Check] Only owner
+            self.ownable.assert_only_owner();
+            // [Check] Router is not zero
+            assert(!router_address.is_zero(), errors::REGISTERING_ROUTER_ZERO);
+
+            // [Check] Valid asset class
+            let asset_class: AssetClass = match asset_class_id.try_into() {
+                Option::Some(a) => a,
+                Option::None(()) => panic_with_felt252(errors::UNKNOWN_ASSET_CLASS)
+            };
+
+            // [Check] Asset class router is correct
+            let router = IAssetClassRouterDispatcher { contract_address: router_address };
+            assert(router.get_asset_class_id() == asset_class_id, errors::MISMATCH_ASSET_CLASS_ID);
+
+            // [Effect] Add the asset class router to the storage
+            self.asset_class_routers.entry(asset_class).write(router);
+
+            // [Interaction] Storage updated event
+            self
+                .emit(
+                    AssetClassRouterAdded {
+                        sender: get_caller_address(),
+                        asset_class_id: asset_class_id,
```

Comment:
We can also but I think that if we return an Enum like that it will actually print the enum encoded to felt, right ? So basically the same as the id. Then I prefer it to be explicit and mark it as `asset_class_id`